sdk/ml/azure-ai-ml/azure/ai/ml/operations/_component_operations.py

```diff
@@ -640,13 +679,20 @@ def _resolve_dependencies_for_pipeline_component_jobs(
             component,
             extra_operations=[
                 self._set_default_display_name_for_anonymous_component_in_node,
+                partial(self._try_resolve_node_level_task_for_parallel_node, resolver=resolver),
             ]
         )
 
         # cache anonymous component only for now
         # request level in-memory cache can be a better solution for other type of assets as they are
         # relatively simple and of small number of distinct instances
-        component_cache = CachedNodeResolver(resolver=resolver)
+        component_cache = CachedNodeResolver(
```

Comment:
Each time when `_resolve_dependencies_for_pipeline_component_jobs` called, a `CachedNodeResolver` object will be created. Will there be issue since the lock object created multiple instances? e.g. when 2 different subgraph using same component

Reply:
Good catch! Let me reconsider where to claim the lock. Maybe just make the lock global